roles/mongodb/tasks/mongodb_auth.yml

```diff
@@ -0,0 +1,68 @@
+---
+- name: Ensure pymongo is installed for the mongodb_user ansible module
+  become: yes
+  pip:
+    name: pymongo
+
+- name: See if authorization is enabled in mongod.conf
+  check_mode: yes
+  lineinfile:
+    path: /etc/mongod.conf
+    insertafter: 'security:'
+    line: '  authorization: enabled'
+  register: _mongo_authorization
+  # changed = line not in file, authorization is disabled
+  # succeeded = line in file, authorization is enabled
+
+- name: Add mongo admin
+  mongodb_user:
+    state: present
+    update_password: on_create
+
+    name: "{{ mongodb_admin_username }}"
+    password: "{{ mongodb_admin_password }}"
+    database: "{{ mongodb_admin_db }}"
+    roles: userAdminAnyDatabase
+
+    login_host: "{{ mongodb_host }}"
+    login_port: "{{ mongodb_port }}"
+    login_user: "{{ _mongo_authorization|changed | ternary(omit, mongodb_admin_username) }}"
+    login_password: "{{ _mongo_authorization|changed | ternary(omit, mongodb_admin_password) }}"
+
+- name: Enable security section in mongod.conf
+  become: yes
+  replace:
+    path: /etc/mongod.conf
+    regexp: '^#security:'
+    replace: 'security:'
+
+- name: Enable authentication in mongod.conf
+  become: yes
+  lineinfile:
+    path: /etc/mongod.conf
+    insertafter: '^security:'
+    line: '  authorization: enabled'
+  notify: restart mongod
+
+# Restart mongod if required here, so that we can safely assume that auth is already enabled when adding more users.
+- meta: flush_handlers
```

Comment:
I would highly recommend to avoid `flush_handlers`, since it will trigger tasks in all another roles we're running.

Reply:
Here's hope for a better way to do this in the future: https://github.com/ansible/ansible/pull/25573

Reply:
Replaced this with an explicitly restart mongodb task if the previous task changed something. I left a TODO to revisit this when flush_handlers can take a filter.